src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/GCCoverFactory.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+internal sealed class GCCoverFactory : IContractFactory<IGCCover>
+{
+    IGCCover IContractFactory<IGCCover>.CreateContract(Target target, int version)
+    {
+        return version switch
```

Comment:
I like how you went for the `switch` despite only having an `if/else` condition to have it ready for the likely new versions in the future 🙂

Reply:
I can't really take credit for this. Aleksey used this factory pattern in the other contracts a few months ago. I do agree its a good approach. 😄